GLOSSARY.md

```diff
@@ -0,0 +1,284 @@
+allow users to inspect JavaScript stack frames as well as the frozen object
+graph in memory (including objects pending garbage collection.)
+
+### Stack Frame
+
+A data structure held in the stack segment of memory representing a function
+call. This typically saves the arguments with which the function was called
+as well as a return address to jump to when the call completes.
+
+### Unwinding the Stack / Top of Stack
+
+Stack frames are collected into a stack that grows and shrinks as functions
+are called and functions return. Exceptions work by "unwinding" the stack
+to the nearest exception handler, if any — that is, they remove stack frames
+giving control back to the nearest frame with an exception handler. The
+stack is grown and shrunk in-place, so any subsequent function calls (even
+ones implicitly made by the JS VM) will clobber the original stack leading
+to the error. `Error` stacktraces _record_ some frame information leading to
+the error, but throwing an error is generally a lossy operation.
+
+If no exception handler is registered in a stack, instead of unwinding the
+stack, V8 may abort the program or call a user-defined `OnUncaughtException`
+handler.
+
+## Promise
+
+The 30-second version:
+
+A [well-specified][promises-aplus] pattern that provides a container type
+designed to describe the dependence graph of asynchronous operations within a
+program. As a container type, any value `T` may be cast to an asynchronous
+value `Promise<T>`. The process of casting `T` to `Promise<T>` is known as
+"resolution". Any value `Promise<T>` may be unwrapped to `T` by passing a
+handler (defined below) to `.then`. Unwrapping via `.then` creates a new
+`Promise` that will resolve to the return value of the handler. One promise
+may have zero-to-many dependent promises.
+
+Errors will propagate from a source promise to all child promises. This process
+is known as "rejection". If a rejected promise has no children, it is
+considered an unhandled rejection. It may be handled in a subsequent operation.
+If a promise has an error handler, propagation from source promises will halt
+at that promise. If a promise has many children, and only one handles errors,
+the other children will still be reported as unhandled rejections.
+
+Unwrapping `Promise<Promise<T>>` will automatically unwrap to `T`. That is:
+`.then(() => Promise.resolve(3)).then(val => console.log(val))` will
+output `3`.
+
+Internally, a promise is represented by a state machine that may be in one of
+three states — pending, resolved, or rejected. The resolved and rejected states
+are collectively referred to as "settled." Once a promise is settled, it cannot
+transition back into pending state.
+
+### Executor / Resolver
+
+The function passed to the `new Promise` constructor. Run **synchronously**.
+Not in common direct use by most promise-users — used mainly to wrap sources of
+asynchrony. As such, it rarely resolves or rejects on the same tick.
```

Comment:
I've seen it a lot in the promise-based code I've used, but it may not be immediately relevant to the discussion in this repo. (In any case, that's less for me to write! :smile:)